packages/language-server/src/check.ts

```diff
@@ -145,7 +146,13 @@ export class AstroCheck {
 		const services = [...createTypeScriptServices(this.ts), createAstroService(this.ts)];
 
 		if (tsconfigPath) {
-			this.linter = kit.createTypeScriptChecker(languagePlugins, services, tsconfigPath);
+			const includeProjectReference = false; // #920
```

Comment:
@Princesseuh To be consistent with old behavior, a false value is used here, but fixing #920 requires changing to true. Since this will impact performance, you may want to pass a flag to decide whether to enable it.